src/Functions/custom/Update-AzFunctionAppPlan.ps1

```diff
@@ -0,0 +1,283 @@
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Path')]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        ${ResourceGroupName},
+
+        [Parameter(Mandatory=$true, ParameterSetName='ByName', HelpMessage='Name of the App Service plan.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Path')]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        ${Name},
+
+        [Parameter(ParameterSetName='ByObjectInput', Mandatory=$true, ValueFromPipeline=$true)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Models.Api20190801.IAppServicePlan]
+        [ValidateNotNull()]
+        ${InputObject},
+
+        [Parameter(HelpMessage='The plan sku. Valid inputs are: EP1, P2, EP3')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Body')]
+        [ArgumentCompleter([Microsoft.Azure.PowerShell.Cmdlets.Functions.Support.SkuType])]
+        [ValidateNotNullOrEmpty()]
+        [System.String]
+        # Sku (EP1, EP2 or EP3)
+        ${Sku},
+
+        [Parameter(HelpMessage='The maximum number of workers for the app service plan.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Body')]
+        [System.Int32]
+        [ValidateRange(1,100)]
+        [Alias("MaxBurst")]
+        ${MaximumWorkerCount},
+
+        [Parameter(HelpMessage='The minimum number of workers for the app service plan.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Body')]
+        [System.Int32]
+        [Alias("MinInstances")]
+        [ValidateRange(1,20)]
+        ${MinimumWorkerCount},
+
+        [Parameter(HelpMessage='Resource tags.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Body')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Runtime.Info(PossibleTypes=([Microsoft.Azure.PowerShell.Cmdlets.Functions.Models.Api20190801.IResourceTags]))]
+        [System.Collections.Hashtable]
+        [ValidateNotNull()]
+        ${Tag},
+
+        [Parameter(HelpMessage='Run the command asynchronously.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        ${NoWait},
+
+        [Parameter(HelpMessage='Run the command as a job.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        ${AsJob},
+
+        [Parameter()]
+        [Alias('AzureRMContext', 'AzureCredential')]
+        [ValidateNotNull()]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Azure')]
+        [System.Management.Automation.PSObject]
+        ${DefaultProfile},
+        
+        [Parameter(DontShow)]
+        [Microsoft.Azure.PowerShell.Cmdlets.Functions.Category('Runtime')]
+        [System.Management.Automation.SwitchParameter]
+        # Wait for .NET debugger to attach
+        ${Break},
+
+        <#
+        [Parameter(DontShow)]
+        [ValidateNotNull()]
```

Comment:
These two parameters can not be removed unless the cmdlet is a pure client side cmdlet(Do not call any Azure services). Without these two parameters, especially ${HttpPipelinePrepend}, you will not be able to run test for these cmdlets in either playback/record mode. See generated\runtime\HttpPipelineMocking.ps1 for details.

Reply:
@dolauli, I removed these two parameters because the generator does not create a type adapter which is needed in order to support -AsJob. Adding @markcowl to provide more context.